taipy/core/_scheduler/_scheduler.py

```diff
@@ -195,3 +204,38 @@ def _update_job_config(cls):
             cls._dispatcher = _StandaloneJobDispatcher()
         else:
             cls._dispatcher = _DevelopmentJobDispatcher()
+
+    @classmethod
+    def _set_process_in_scheduler(cls, job_id, process):
+        cls._processes[job_id] = process
+
+    @classmethod
+    def _pop_process_in_scheduler(cls, job_id, default=None):
+        return cls._processes.pop(job_id, default)  # type: ignore
+
+    @classmethod
+    def _cancel_job(cls, job):
+        if process := cls._pop_process_in_scheduler(job.id):
```

Comment:
What about retrieving directly the submit_id here and cancel all the jobs with the same submit_id ? WOuld it make the design easier?